src/algebra/ordered_group.lean

```diff
@@ -98,341 +89,401 @@ instance ordered_comm_group.has_exists_mul_of_le (α : Type u)
-begin
-  have h := mul_lt_mul_left' h a,
-  rwa mul_inv_cancel_left at h
-end
+@[simp, to_additive]
+lemma mul_inv_le_iff_le_mul : a * b⁻¹ ≤ c ↔ a ≤ c * b :=
+by rw [← mul_le_mul_iff_right b, inv_mul_cancel_right]
 
-@[to_additive]
-lemma lt_inv_mul_of_mul_lt (h : a * b < c) : b < a⁻¹ * c :=
-begin
-  have h := mul_lt_mul_left' h (a⁻¹),
-  rwa inv_mul_cancel_left at h
-end
+@[simp, to_additive]
+lemma inv_mul_le_one_iff_le : a * b⁻¹ ≤ 1 ↔ a ≤ b :=
+by rw [← mul_le_mul_iff_right b, inv_mul_cancel_right, one_mul]
 
-@[to_additive]
-lemma lt_mul_of_inv_mul_lt (h : b⁻¹ * a < c) : a < b * c :=
-begin
-  have h := mul_lt_mul_left' h b,
-  rwa mul_inv_cancel_left at h
-end
+end typeclasses_right_le
 
-@[to_additive]
-lemma inv_mul_lt_of_lt_mul (h : a < b * c) : b⁻¹ * a < c :=
-begin
-  have h := mul_lt_mul_left' h (b⁻¹),
-  rwa inv_mul_cancel_left at h
-end
+section typeclasses_right_lt
+variables [has_lt α] [covariant_class α α (function.swap (*)) (<)] {a b c : α}
 
-@[to_additive]
-lemma lt_mul_of_inv_mul_lt_left (h : b⁻¹ * a < c) : a < b * c :=
-lt_mul_of_inv_mul_lt h
+/--  Uses `right` co(ntra)variant. -/
+@[simp, to_additive right.neg_neg_iff]
+lemma right.inv_lt_one_iff :
+  a⁻¹ < 1 ↔ 1 < a :=
+by rw [← mul_lt_mul_iff_right a, inv_mul_self, one_mul]
 
-@[to_additive]
-lemma inv_mul_lt_left_of_lt_mul (h : a < b * c) : b⁻¹ * a < c :=
-inv_mul_lt_of_lt_mul h
+/--  Uses `right` co(ntra)variant. -/
+@[simp, to_additive right.neg_pos_iff]
+lemma right.one_lt_inv_iff :
+  1 < a⁻¹ ↔ a < 1 :=
+by rw [← mul_lt_mul_iff_right a, inv_mul_self, one_mul]
 
 @[to_additive]
-lemma lt_mul_of_inv_mul_lt_right (h : c⁻¹ * a < b) : a < b * c :=
-by { rw mul_comm, exact lt_mul_of_inv_mul_lt h }
+lemma inv_lt_iff_one_lt_mul : a⁻¹ < b ↔ 1 < b * a :=
+(mul_lt_mul_iff_right a).symm.trans $ by rw inv_mul_self
 
 @[to_additive]
-lemma inv_mul_lt_right_of_lt_mul (h : a < b * c) : c⁻¹ * a < b :=
-by { rw mul_comm at h, exact inv_mul_lt_of_lt_mul h }
+lemma lt_inv_iff_mul_lt_one : a < b⁻¹ ↔ a * b < 1 :=
+(mul_lt_mul_iff_right b).symm.trans $ by rw inv_mul_self
 
 @[simp, to_additive]
-lemma inv_lt_one_iff_one_lt : a⁻¹ < 1 ↔ 1 < a :=
-⟨ one_lt_of_inv_inv, inv_inv_of_one_lt ⟩
+lemma mul_inv_lt_iff_lt_mul : a * b⁻¹ < c ↔ a < c * b :=
+by rw [← mul_lt_mul_iff_right b, inv_mul_cancel_right]
+
+@[simp, to_additive]
+lemma inv_mul_lt_one_iff_lt : a * b⁻¹ < 1 ↔ a < b :=
+by rw [← mul_lt_mul_iff_right b, inv_mul_cancel_right, one_mul]
+
+end typeclasses_right_lt
+
+section typeclasses_left_right_le
+variables [has_le α] [covariant_class α α (*) (≤)] [covariant_class α α (function.swap (*)) (≤)]
+  {a b c d : α}
 
 @[simp, to_additive]
 lemma inv_le_inv_iff : a⁻¹ ≤ b⁻¹ ↔ b ≤ a :=
-have a * b * a⁻¹ ≤ a * b * b⁻¹ ↔ a⁻¹ ≤ b⁻¹, from mul_le_mul_iff_left _,
-by { rw [mul_inv_cancel_right, mul_comm a, mul_inv_cancel_right] at this, rw [this] }
+by { rw [← mul_le_mul_iff_left a, ← mul_le_mul_iff_right b], simp }
+
+alias neg_le_neg_iff ↔ le_of_neg_le_neg _
```

Comment:
Yes, most of the times (probably all, I do not remember), I reformulated the statement from the additive version to multiplicative with `to_additive`, and then I only introduced the alias for the additive version, since the multiplicative version was not in Lean before anyway.